.github/workflows/test.yml

```diff
@@ -29,60 +29,83 @@ jobs:
         run: ~/shfmt -d -i 2 -s test/testlib.sh.in test/*.test
 
   linux:
+    runs-on: ubuntu-latest
     strategy:
-      matrix:
-        os: [ubuntu-18.04, ubuntu-20.04]
-        compiler: [clang, gcc]
-        legacy_protocol: ["", --disable-legacy-protocol]
       fail-fast: false
-
-    runs-on: ${{ matrix.os }}
-    env:
-      CC: ${{ matrix.compiler }}
+      matrix:
+        os:
+          - alpine
+          - centos:7 # aka RHEL 7
+          - almalinux:8 # aka RHEL 8
+          - debian:oldstable
+          - debian:stable
```

Comment:
Is there a debian:testing? If so, that would be great to add.

Reply:
Sure, I'll add testing and sid to the mix. I was afraid you say that there are too many images already.

Reply:
Maybe. But Debian stable can be quite old at times, having testing is helpful. If it's too much, I'd rather keep Debian oldstable+stable+testing to ensure we can build against old and new versions of libraries, and Alpine because it uses musl and is quite different from the other distros.